lib/engines/content_object_store/app/models/content_object_store/result_monad.rb

```diff
@@ -0,0 +1,11 @@
+module ContentObjectStore
+  ResultMonad = Struct.new(:success, :message, :object) do
```

Comment:
Would it be better to use `Data.define` here? So:

```suggestion
  ResultMonad < Data.define(:success, :message, :object)
```

That way the monad behaves more like a Class (also I think Structs are historically quite slow)

Reply:
Interestingggggg! [This definitely looks](https://github.com/ruby/ruby/pull/6353) like a more modern way to do it (Couldn't find anything on speed improvements). We also get named params this way which I much prefer and would level this up.

Reply:
Done 👍

Reply:
Ah, I was getting Structs mixed up with OpenStructs, which are veeeery slow - https://www.honeybadger.io/blog/how-openstruct-and-hashes-can-kill-performance/